py4web/core.py

```diff
@@ -953,20 +955,42 @@ def error404(error):
 # Web Server and Reload Logic: Operations
 #########################################################################################
 
+def watch_folder_event_loop(apps_folder):
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    loop.run_until_complete(watch_folder(apps_folder))
+
+async def watch_folder(apps_folder):
+    click.echo('watching python file changes in: %s' % apps_folder)
+    async for changes in awatch(os.path.join(apps_folder)):
+        for app in set(map(lambda p: p.relative_to(apps_folder).parts[0],
+                        filter(lambda p: p.suffix == '.py',
+                                map(lambda pair: pathlib.Path(pair[1]), changes)))):
+            Reloader.import_app(app)
 
 def start_server(args):
-    host, port = args['host'], int(args['port'])
+    host, port, apps_folder, watch = args['host'], int(args['port']), args['apps_folder'], args['watch']
+
     if platform.system().lower() == "windows":
+        if watch:
+            # default wsgi server block the main thread so we open a new thread for the file watcher
+            threading.Thread(target=watch_folder_event_loop, args=(apps_folder,)).start()
         # Tornado fail on windows
         bottle.run(host=host, port=int(port), reloader=False)
     elif args['number_workers'] < 1:
+        if watch:
+            # tornado delegate to asyncio so we add a future into the event loop
+            asyncio.ensure_future(watch_folder(apps_folder))
         bottle.run(server="tornado", host=host, port=port, reloader=False)
     else:
         if not gunicorn:
             logging.error("gunicorn not installed")
         elif not gevent:
             logging.error("gevent not installed")
         else:
+            if watch:
```

Comment:
Actually, I want to point out here I didn't add the --watch ability when args['number_workers'] > 0 because I think users only use the number_workers option to deal with high server load in production, which is contracting the usage context of `--watch`